pkg/types/ibmcloud/validation/platform.go

```diff
@@ -58,5 +63,68 @@ func ValidatePlatform(p *ibmcloud.Platform, fldPath *field.Path) field.ErrorList
 	if p.DefaultMachinePlatform != nil {
 		allErrs = append(allErrs, ValidateMachinePool(p, p.DefaultMachinePlatform, fldPath.Child("defaultMachinePlatform"))...)
 	}
+
+	if p.ServiceEndpoints != nil {
+		allErrs = append(allErrs, validateServiceEndpoints(p.ServiceEndpoints, fldPath.Child("serviceEndpoints"))...)
+	}
+	return allErrs
+}
+
+// validateServiceEndpoints checks that the specified ServiceEndpoints are valid.
+func validateServiceEndpoints(endpoints []configv1.IBMCloudServiceEndpoint, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	tracker := map[string]int{}
+	for index, endpoint := range endpoints {
+		fldp := fldPath.Index(index)
+		if eindex, ok := tracker[endpoint.Name]; ok {
+			allErrs = append(allErrs, field.Invalid(fldp.Child("name"), endpoint.Name, fmt.Sprintf("duplicate service endpoint not allowed for %s, service endpoint already defined at %s", endpoint.Name, fldPath.Index(eindex))))
+		} else {
+			tracker[endpoint.Name] = index
+		}
+
+		if err := validateServiceURL(endpoint.URL); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldp.Child("url"), endpoint.URL, err.Error()))
+		}
+	}
 	return allErrs
 }
+
+// schemeRE is used to check whether a string starts with a scheme (URI format).
+var schemeRE = regexp.MustCompile("^([^:]+)://")
+
+// validateServiceURL checks that a string meets certain URI expectations.
+func validateServiceURL(uri string) error {
+	endpoint := uri
+	httpsScheme := "https"
+
+	// Regexp for a trailing API version in URL path ('/v1', '/v22/', etc.)
+	versionPath := regexp.MustCompile(`(/v\d+[/]{0,1})$`)
```

Comment:
would it make sense to compile this regex once as in line 93?

Reply:
Oh yeah, I can move that.

Reply:
Moved